[nativert] Move KernelFactory to PyTorch core

henryoier opened the pull request:
Summary: Kernel factory handles the kernel nodes initializations and different type of kernels executions.

Test Plan:
CI

Rollback Plan:

Differential Revision: D77346836

henryoier commented:
@pytorchbot label "topic: not user facing

henryoier commented:
@pytorchbot label "topic: not user facing"

zhxchen17 commented on torch/nativert/kernels/KernelFactory.cpp:
```diff
@@ -0,0 +1,270 @@
+#include <c10/util/string_view.h>
+#include <fmt/ranges.h>
+
+#include <torch/nativert/executor/DelegateExecutor.h>
+#include <torch/nativert/executor/OpKernel.h>
+#include <torch/nativert/executor/ParallelGraphExecutor.h>
+#include <torch/nativert/executor/SerialGraphExecutor.h>
+#include <torch/nativert/graph/Graph.h>
+#include <torch/nativert/kernels/AutoFunctionalizeKernel.h>
+#include <torch/nativert/kernels/C10Kernel.h>
+#include <torch/nativert/kernels/CallTorchBindKernel.h>
+#include <torch/nativert/kernels/HigherOrderKernel.h>
+#include <torch/nativert/kernels/KernelFactory.h>
+#include <torch/nativert/kernels/PrimKernelRegistry.h>
```
according to the linter, headers need to be sorted.

henryoier committed:
[nativert] Move KernelFactory to PyTorch core (#156913)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/156913

Kernel factory handles the kernel nodes initializations and different type of kernels executions.

Test Plan:
CI

Rollback Plan:

Reviewed By: zhxchen17

Differential Revision: D77346836
